diff --git a/src/main/java/org/apache/commons/lang3/concurrent/AbstractCircuitBreaker.java b/src/main/java/org/apache/commons/lang3/concurrent/AbstractCircuitBreaker.java
index 20891c0ff..e6db18825 100644
--- a/src/main/java/org/apache/commons/lang3/concurrent/AbstractCircuitBreaker.java
+++ b/src/main/java/org/apache/commons/lang3/concurrent/AbstractCircuitBreaker.java
@@ -121,9 +121,15 @@ protected void changeState(final State newState) {
 
     /**
      * {@inheritDoc}
+     * <p>
+     * This implementation checks the state of the circuit breaker and returns true if it is closed.
+     * Subclasses can override this method to provide additional state checking logic.
+     * </p>
      */
     @Override
-    public abstract boolean checkState();
+    public boolean checkState() {
+        return !isOpen();
+    }
 
     /**
      * {@inheritDoc}
diff --git a/src/main/java/org/apache/commons/lang3/concurrent/CircuitBreakingException.java b/src/main/java/org/apache/commons/lang3/concurrent/CircuitBreakingException.java
index da3a1e331..0b0a0dd0f 100644
--- a/src/main/java/org/apache/commons/lang3/concurrent/CircuitBreakingException.java
+++ b/src/main/java/org/apache/commons/lang3/concurrent/CircuitBreakingException.java
@@ -16,13 +16,20 @@
  */
 package org.apache.commons.lang3.concurrent;
 
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.lang3.exception.ContextedRuntimeException;
+import org.apache.commons.lang3.exception.ExceptionContext;
+import org.apache.commons.lang3.tuple.Pair;
+
 /**
  * An exception class used for reporting runtime error conditions related to
  * circuit breakers.
  *
  * @since 3.5
  */
-public class CircuitBreakingException extends RuntimeException {
+public class CircuitBreakingException extends ContextedRuntimeException {
 
     /**
      * The serial version UID.
@@ -33,6 +40,7 @@ public class CircuitBreakingException extends RuntimeException {
      * Creates a new, uninitialized instance of {@link CircuitBreakingException}.
      */
     public CircuitBreakingException() {
+        super();
     }
 
     /**
@@ -63,4 +71,15 @@ public CircuitBreakingException(final Throwable cause) {
         super(cause);
     }
 
+    /**
+     * Instantiates CircuitBreakingException with cause, message, and ExceptionContext.
+     *
+     * @param message  the exception message, may be null
+     * @param cause  the underlying cause of the exception, may be null
+     * @param context  the context used to store the additional information, null uses default implementation
+     */
+    public CircuitBreakingException(final String message, final Throwable cause, ExceptionContext context) {
+        super(message, cause, context);
+    }
+
 }
diff --git a/src/main/java/org/apache/commons/lang3/concurrent/ThresholdCircuitBreaker.java b/src/main/java/org/apache/commons/lang3/concurrent/ThresholdCircuitBreaker.java
index 6b7d8719c..30300c576 100644
--- a/src/main/java/org/apache/commons/lang3/concurrent/ThresholdCircuitBreaker.java
+++ b/src/main/java/org/apache/commons/lang3/concurrent/ThresholdCircuitBreaker.java
@@ -77,14 +77,6 @@ public ThresholdCircuitBreaker(final long threshold) {
         this.threshold = threshold;
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public boolean checkState() {
-        return !isOpen();
-    }
-
     /**
      * {@inheritDoc}
      *
